src/ItemManager.vala

```diff
@@ -129,18 +131,30 @@
         }
     }
 
-    private void add_launcher (Launcher launcher, int index = -1) {
-        launcher.removed.connect (remove_launcher);
+    private void setup_item_removed (BaseItem item) {
+        item.removed.connect ((_item) => {
+            if (_item is Launcher) {
+                launchers.remove ((Launcher) _item);
```

Comment:
If we keep doing this in `remove_item` we wouldn't need this extra method would we? Instead we could keep the one liner signal connection :)
Or am I missing some context needed for the icon groups?

Reply:
Thanks for looking through this, this method is really not needed. Removed it.